workers/unity/Packages/com.improbable.gdk.querybasedinteresthelper/InterestHelper.cs

```diff
@@ -0,0 +1,220 @@
+using System.Collections.Generic;
+using System.Linq;
+using Improbable.Gdk.Core;
+
+namespace Improbable.Gdk.QueryBasedInterest
+{
+    /// <summary>
+    ///     Utility class to help construct Interest component snapshots.
+    /// </summary>
+    public class InterestHelper
```

Comment:
This class generally looks very useful, although it might be worth comparing it to the planned extensions to the EntityTemplate to see if we can use the same pattern again. Broadly speaking that you have some base thing you create, then you create a new instance of it that makes a copy but with some modification. Rather than modifying the object directly. Don't know if that pattern is actually better but it might be worth a look.